tests/functional/metadata/mongodb/listObject.spec.js

```diff
@@ -0,0 +1,306 @@
+                FULL_CONTROL: [],
+                WRITE: [],
+                WRITE_ACP: [],
+                READ: [],
+                READ_ACP: [],
+            },
+            _mdBucketModelVersion: 10,
+            _transient: false,
+            _deleted: false,
+            _serverSideEncryption: null,
+            _versioningConfiguration: null,
+            _locationConstraint: 'us-east-1',
+            _readLocationConstraint: null,
+            _cors: null,
+            _replicationConfiguration: null,
+            _lifecycleConfiguration: null,
+            _uid: '',
+            _isNFS: null,
+            ingestion: null,
+        });
+        const versionParams = {
+            versioning: true,
+            versionId: null,
+            repairMaster: null,
+        };
+        async.series([
+            next => metadata.createBucket(BUCKET_NAME, bucketMD, logger, err => {
+                if (err) {
+                    return next(err);
+                }
+                return next();
+            }),
+            next => {
+                const params = {
+                    objName: 'pfx1-test-object',
+                    objVal: {
+                        key: 'pfx1-test-object',
+                        versionId: 'null',
+                    },
+                    nbVersions: 5,
+                };
+                putBulkObjectVersions(BUCKET_NAME, params.objName, params.objVal, versionParams,
+                params.nbVersions, next);
+            },
+            next => {
+                const params = {
+                    objName: 'pfx2-test-object',
+                    objVal: {
+                        key: 'pfx2-test-object',
+                        versionId: 'null',
+                    },
+                    nbVersions: 5,
+                };
+                putBulkObjectVersions(BUCKET_NAME, params.objName, params.objVal, versionParams,
+                params.nbVersions, next);
+            },
+            next => {
+                const params = {
+                    objName: 'pfx3-test-object',
+                    objVal: {
+                        key: 'pfx3-test-object',
+                        versionId: 'null',
+                    },
+                    nbVersions: 5,
+                };
+                putBulkObjectVersions(BUCKET_NAME, params.objName, params.objVal, versionParams,
+                params.nbVersions, next);
+            },
+        ], done);
+    });
+
+    afterEach(done => {
+        metadata.deleteBucket(BUCKET_NAME, logger, done);
+    });
+
+    it('Should list master versions of objects', done => {
+        const bucketName = BUCKET_NAME;
+        const params = {
+            listingType: 'DelimiterMaster',
+            maxKeys: 100,
+        };
+
+        return metadata.listObject(bucketName, params, logger, (err, data) => {
+            assert.deepStrictEqual(err, null);
+            assert.strictEqual(data.Contents.length, 3);
```

Comment:
should also check the keys of the returned items?
(for all these listing tests)